air/src/execution_step/value_types/stream/stream_definition.rs

```diff
@@ -551,4 +574,32 @@ mod test {
             ))
         ));
     }
+
+    #[test]
+    fn stream_size_limit() {
+        use super::STREAM_MAX_SIZE;
+        use crate::UncatchableError;
+
+        let mut stream = Stream::new();
+
+        let value = create_value(json!("1"));
+
+        for _ in 0..STREAM_MAX_SIZE / 2 {
+            stream.add_value(value.clone(), Generation::current(0)).unwrap();
+        }
+
+        for _ in 0..STREAM_MAX_SIZE / 4 {
+            stream.add_value(value.clone(), Generation::previous(0)).unwrap();
+        }
+
+        for _ in 0..STREAM_MAX_SIZE / 4 - 1 {
+            stream.add_value(value.clone(), Generation::new()).unwrap();
+        }
+
+        let add_value_result = stream.add_value(value.clone(), Generation::new());
+
+        assert!(matches!(add_value_result, Err(ExecutionError::Uncatchable(_))));
+        let Err(ExecutionError::Uncatchable(error)) = add_value_result else { panic!("there must be CatchableError")};
```

Comment:
UncatchableError?

Reply:
Let's even remove this check because the following line covers it

Reply:
If I make it catchable is not enforced. Are you sure we want it Catchable?

Reply:
This is not a check but a decomposition to get to Uncatchabe enum value.